### ISSUE TITLE
[feature] Provide a `kfp.components.load_component_from_dict` function

### ISSUE THREAD

**Issue by omarzouk** [kubeflow/pipelines#5708]
### Feature Area

/area sdk

### What feature would you like to see?

Provide a `load_component_from_dict`  function that calls `ComponentSpec.from_dict` under the hood, which would allw the user to create operators in a fashion similar to the old `dsl.ContainerOp` without having to generate an intermediate yaml file. The spec of this dictionary can be 1-to-1 with the yaml and this would be easy to maintain.

### What is the use case or pain point?

Forcing a the use of yaml files on the SDK users is an arbitrary decision/choice, especially if they are already using a fully featured programming language like python

### Is there a workaround currently?

https://github.com/kubeflow/pipelines/issues/3748#issuecomment-627698554 however it seems to be using the internals of the SDK which means its not future proof or safe to use.

Love this idea? Give it a 👍. We prioritize fulfilling features with the most 👍.

**Comment by Ark-kun** [comment]
Would a `load_component_spec_from_component_spec(ComponentSpec)` function be desirable as a replacement?

**Comment by omarzouk** [comment]
> Would a `load_component_spec_from_component_spec(ComponentSpec)` function be desirable as a replacement?

yes absolutely, as long as this way of building containers is guaranteed to be maintained in the future, it is a pain in the ass to have to re-write all components just to describe basic properties such as image name and inputs and outputs.
